src/frontend/parser.mly

```diff
@@ -16,6 +16,20 @@ let build_id id loc =
   grammar_logger ("identifier " ^ id);
   {name=id; id_loc=location_span_of_positions loc}
 
+let reserved id =
+  raise (Errors.SyntaxError (Errors.Parsing
+          ("Expected a new identifier but found reserved keyword '" ^ id.name ^ "'.\n",
+            id.id_loc)))
+
+let reserved_decl (id, is_type) =
+  if is_type then
+    raise (Errors.SyntaxError (Errors.Parsing
+          ("Found a type ('"^id.name^
+            "') where an identifier was expected.\nAll variables declared in a comma-separated list must be of the same type.\n",
```

Comment:
That `^id.name^` needs space and the following line is too long. Since ocamlformat doesn't recognize this `.mly` file I usually copy-paste the modified functions to a random `.ml` file, format that file, and then copy-paste the formatted functions back here.